### PR TITLE
SONARPY-1139 ApiGateway: RestApi part 2

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java [hunk 1]
```diff
@@ -19,30 +19,97 @@
...
+
+  private static Optional<String> enclosingMethodFqn(Tree tree) {
+    FunctionDef functionDef = (FunctionDef) TreeUtils.firstAncestorOfKind(tree, Tree.Kind.FUNCDEF);
+    return Optional.ofNullable(functionDef)
+      .map(FunctionDefImpl.class::cast)
+      .map(FunctionDefImpl::functionSymbol)
+      .map(Symbol::fullyQualifiedName);
+  }
+
+  private static boolean isSafe(SubscriptionContext subscriptionContext, CdkUtils.ExpressionFlow argument) {
```
Comment (nils-werner-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java:92]: Maybe add an entity name about what exactly is safe. You store the fqn in a field called `safeMethods` maybe `isSafeMethod` is a proper name?

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! See my comments

**Review by nils-werner-sonarsource — COMMENTED** [review]
Do you think of a test case that covers the missing 3%?

**Comment by mstachniuk** [comment]
I fixed the coverage

File: python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java [hunk 2]
```diff
@@ -19,30 +19,100 @@
...
+          }
+        }
+      )
+    );
     checkFqn("aws_cdk.aws_apigateway.Resource.add_method", (subscriptionContext, callExpression) ->
-      getArgument(subscriptionContext, callExpression, "authorization_type").ifPresent(
-        argument -> argument.addIssueIf(isFqn("aws_cdk.aws_apigateway.AuthorizationType.NONE"), MESSAGE)
+      getArgument(subscriptionContext, callExpression, AUTHORIZATION_TYPE).ifPresentOrElse(
+        argument -> argument.addIssueIf(isFqn(AUTHORIZATION_TYPE_NONE), MESSAGE),
+        () -> enclosingMethodFqn(callExpression).filter(fqn -> safeMethods.contains(fqn)).ifPresentOrElse(fqn -> {}, () ->
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java]: It took me a moment to understand what you were trying to do here and why.

**Pull request merged**